nautobot/dcim/factory.py

```diff
@@ -0,0 +1,146 @@
+import factory
+import pytz
+from factory.django import DjangoModelFactory
+
+from django.contrib.contenttypes.models import ContentType
+
+from nautobot.dcim.models import Location, LocationType, Region, Site, Device
+from nautobot.extras.models import Status
+from nautobot.ipam.models import Prefix, VLAN, VLANGroup
+from nautobot.tenancy.models import Tenant
+from nautobot.utilities.factory import random_instance, UniqueFaker
+
+
+class RegionFactory(DjangoModelFactory):
+    class Meta:
+        model = Region
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    has_parent = factory.Faker("pybool")
+    parent = factory.Maybe("has_parent", random_instance(Region), None)
+    name = factory.Maybe("has_parent", UniqueFaker("city"), UniqueFaker("country"))
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+
+class SiteFactory(DjangoModelFactory):
+    class Meta:
+        model = Site
+        exclude = ("has_region", "has_tenant")
+
+    name = UniqueFaker("street_address")
+    status = random_instance(lambda: Status.objects.get_for_model(Site), allow_null=False)
+
+    has_region = factory.Faker("pybool")
+    region = factory.Maybe("has_region", random_instance(Region), None)
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+
+    time_zone = pytz.timezone("US/Eastern")
+    physical_address = factory.Faker("address")
+    shipping_address = factory.Faker("address")
+    latitude = factory.Faker("latitude")
+    longitude = factory.Faker("longitude")
+    contact_name = factory.Faker("name")
+    contact_phone = factory.Sequence(lambda n: f"1091-65912-{n:04d}")
```

Comment:
Why not `factory.Faker("phone_number")`?

Reply:
phone number generates more than 20 characters and raises validation errors. And it doesnt seems to have a `max_limit` like country_calling_code.
<img width="866" alt="Screen Shot 2022-10-07 at 11 37 10 AM" src="https://user-images.githubusercontent.com/46973263/194593108-c25abaa6-43ca-4b50-ad2d-0a32c1d4141e.png">

Reply:
Oh geez, I didn't realize we limit our `phone_number` to 20 characters maximum. That seems inadequate. :-(